collector/remote_fx.go

```diff
@@ -0,0 +1,472 @@
+	FECRate                  *prometheus.Desc
+	LossRate                 *prometheus.Desc
+	RetransmissionRate       *prometheus.Desc
+	TCPReceivedRate          *prometheus.Desc
+	TCPSentRate              *prometheus.Desc
+	TotalReceivedRate        *prometheus.Desc
+	TotalSentRate            *prometheus.Desc
+	TotalReceivedBytes       *prometheus.Desc
+	TotalSentBytes           *prometheus.Desc
+	UDPPacketsReceivedPersec *prometheus.Desc
+	UDPPacketsSentPersec     *prometheus.Desc
+	UDPReceivedRate          *prometheus.Desc
+	UDPSentRate              *prometheus.Desc
+
+	//gfx
+	AverageEncodingTime                                *prometheus.Desc
+	FrameQuality                                       *prometheus.Desc
+	FramesSkippedPerSecondInsufficientClientResources  *prometheus.Desc
+	FramesSkippedPerSecondInsufficientNetworkResources *prometheus.Desc
+	FramesSkippedPerSecondInsufficientServerResources  *prometheus.Desc
+	GraphicsCompressionratio                           *prometheus.Desc
+	InputFramesPerSecond                               *prometheus.Desc
+	OutputFramesPerSecond                              *prometheus.Desc
+	SourceFramesPerSecond                              *prometheus.Desc
+}
+
+// NewRemoteFx ...
+func NewRemoteFx() (Collector, error) {
+	const subsystem = "remote_fx"
+	return &RemoteFxCollector{
+		// net
+		BaseTCPRTT: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "net_base_tcp_rrt"),
+			"Base TCP round-trip time (RTT) detected in milliseconds",
+			[]string{"session"},
+			nil,
+		),
+		BaseUDPRTT: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "net_base_udp_rrt"),
+			"Base UDP round-trip time (RTT) detected in milliseconds.",
+			[]string{"session"},
+			nil,
+		),
+		CurrentTCPBandwidth: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "net_current_tcp_bandwidth"),
+			"TCP Bandwidth detected in thousands of bits per second (1000 bps).",
+			[]string{"session"},
+			nil,
+		),
+		CurrentTCPRTT: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "net_current_tcp_rtt"),
+			"Average TCP round-trip time (RTT) detected in milliseconds.",
+			[]string{"session"},
+			nil,
+		),
+		CurrentUDPBandwidth: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "net_current_udp_bandwidth"),
+			"UDP Bandwidth detected in thousands of bits per second (1000 bps).",
+			[]string{"session"},
+			nil,
+		),
+		CurrentUDPRTT: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "net_current_udp_rtt"),
+			"Average UDP round-trip time (RTT) detected in milliseconds.",
+			[]string{"session"},
+			nil,
+		),
+		FECRate: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "net_fec_rate"),
+			"Forward Error Correction (FEC) percentage",
+			[]string{"session"},
+			nil,
+		),
+		LossRate: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "net_loss_rate"),
+			"Loss percentage",
```

Comment:
Is this a loss percentage for all traffic in the session?

Reply:
documentation is not clear but all these counters are /session